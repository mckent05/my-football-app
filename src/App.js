import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Routes, Route } from 'react-router-dom';
import { getLeagues } from './Redux/HomePage/Home';
import HomePage from './Components/Pages/HomePage/HomePage';
import StandingPage from './Components/Pages/StandingPage/StandingPage';
import './App.css';

const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getLeagues());
  }, []);

  return (
    <div className="App">
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/StandingPage/:id" element={<StandingPage />} />
      </Routes>
    </div>
  );
};

export default App;
