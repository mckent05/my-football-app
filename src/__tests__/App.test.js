import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import App from '../App';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('testing the app', () => {
  it('testing HomePage UI', () => {
    act(() => {
      render(
        <React.StrictMode>
          <BrowserRouter>
            <Provider store={store}>
              <App />
            </Provider>
          </BrowserRouter>
        </React.StrictMode>, container,
      );
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
