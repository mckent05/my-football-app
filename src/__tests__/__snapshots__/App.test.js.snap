// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the app testing HomePage UI 1`] = `
"<div class=\\"App\\">
  <div class=\\"home\\">
    <div class=\\"standing-page-header home-page-header d-flex j-center a-center\\">
      <div>
        <h2>Steve's Football App</h2>
      </div>
      <div class=\\"d-flex j-center a-center season-select\\">
        <p>Leagues Displayed:0</p>
      </div>
    </div><input type=\\"text\\" class=\\"search\\" placeholder=\\"search league\\" value=\\"\\">
    <div class=\\"home-page d-flex j-center a-center\\"></div>
  </div>
</div>"
`;
